File: pulse-go/internal/pulse/engine/move_type_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package engine

import (
	"testing"
)

func TestMoveType(t *testing.T) {
	t.Run("MoveTypes array should be valid", func(t *testing.T) {
		for mt := range MoveTypes {
			if MoveTypes[mt] != mt {
				t.Errorf("wanted move types at index %d to be %v, but got %v", mt, mt, MoveTypes[mt])
			}
		}
	})
}

File: pulse-kotlin/src/commonTest/kotlin/com/fluxchess/pulse/kotlin/engine/MoveTypeTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package com.fluxchess.pulse.kotlin.engine

import kotlin.test.Test
import kotlin.test.assertEquals

class MoveTypeTest {
	@Test
	fun `MoveTypes array should be valid`() {
		for (moveType in moveTypes.indices) {
			assertEquals(moveTypes[moveType], moveType)
		}
	}
}
